src/rust/engine/rule_graph/src/lib.rs

```diff
@@ -786,31 +793,85 @@ pub fn params_str<T: TypeId>(params: &ParamTypes<T>) -> String {
   T::display(params.iter().cloned())
 }
 
-///
-/// TODO: Move all of these methods to Display impls.
-///
 pub fn entry_str<R: Rule>(entry: &Entry<R>) -> String {
-  match entry {
-    Entry::WithDeps(ref e) => entry_with_deps_str(e),
-    Entry::Param(type_id) => format!("Param({})", type_id),
+  entry_node_str_with_attrs(entry).entry_str
+}
+
+#[derive(Debug, Clone, PartialOrd, Ord, PartialEq, Eq)]
+pub struct GraphVizEntryWithAttrs {
+  entry_str: String,
+  attrs_str: Option<String>,
+}
+
+pub enum Palette {
+  Olive,
+  Gray,
+  Orange,
+  Blue,
+}
+
+impl Palette {
+  // https://c.eev.ee/kouyou/ is wonderful for selecting lovely color juxtapositions across multiple
+  // different color axes.
+  fn color_string(&self) -> String {
+    // These color values are all in HSV. See https://www.graphviz.org/doc/info/colors.html for
+    // other methods of specifying
+    // colors. https://renenyffenegger.ch/notes/tools/Graphviz/attributes/_color/index may also be
+    // useful.
+    match self {
+      Self::Olive => "0.2214,0.7179,0.8528".to_string(),
+      Self::Gray => "0.576,0,0.6242".to_string(),
+      Self::Orange => "0.08,0.5,0.976".to_string(),
+      Self::Blue => "0.5,1,0.9".to_string(),
+    }
+  }
+
+  pub fn color_format_string(&self) -> String {
+    format!("[color=\"{}\",style=filled]", self.color_string())
+  }
+}
+
+///
+/// Apply coloration to several nodes.
+pub fn entry_node_str_with_attrs<R: Rule>(entry: &Entry<R>) -> GraphVizEntryWithAttrs {
+  let (entry_str, attrs_str) = match entry {
+    &Entry::WithDeps(ref e) => (
+      entry_with_deps_str(e),
+      // Color "singleton" entries (with no params) as a nice olive green!
```

Comment:
I would remove the reference to the color from the comment because it is now self-explanatory and if someone were to change the color later, they would probably forget to update this comment.

If you want to describe the color "Olive" as "A nice olive green", that comment would make more sense at the enum definition level where it would be less likely to be missed in a future code change.

Comment also applies to "Steely gray" and "laid-back orange" down there.

Reply:
Done! Thank you!!